web/src/filter.ts

```diff
@@ -831,7 +832,17 @@ export class Filter {
             });
 
             assert(typeof message.display_recipient !== "string");
-            const dm_participants = message.display_recipient.map((user) => user.email);
+
+            // We should make sure the current user is not included for
+            // the `dm` operand for the narrow.
+            const dm_participants = message.display_recipient
+                .map((user) => user.email)
+                .filter((user_email) => user_email !== current_user.email);
+            // However, if the current user is the only recipient of the
+            // message, we should include the user in the operand.
+            if (dm_participants.length === 0) {
+                dm_participants.push(current_user.email);
+            }
```

Comment:
This is important because without this, the recipients for `dm` narrow would also contain `current_user`. This would matter when you try to get bucketed unread messages -- since the unreads are bucketed with recipients that don't include `current_user`, the unreads returned would be `[]`